File: frontend-admin/src/api/feedback.ts
import http from './http'

export interface FeedbackItem {
  id: number
  content: string
  contact: string | null
  status: 'pending' | 'processing' | 'resolved' | 'closed'
  reply: string | null
  reply_at: string | null
  user_id: number | null
  username: string | null
  created_at: string
  updated_at: string
}

export interface PageResult<T> {
  items: T[]
  total: number
  page: number
  page_size: number
}

export function listFeedback(page = 1, page_size = 20, status?: string): Promise<PageResult<FeedbackItem>> {
  return http.get('/feedback', { params: { page, page_size, status_filter: status || undefined } })
}

export function updateFeedback(id: number, payload: { status?: string; reply?: string }): Promise<FeedbackItem> {
  return http.put(`/feedback/${id}`, payload)
}

File: backend/app/schemas/feedback.py
"""用户反馈相关 Schema。"""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class FeedbackBase(BaseModel):
    content: str = Field(..., min_length=1)
    contact: str | None = None
    status: str = "pending"  # pending / processing / resolved / closed


class FeedbackCreate(BaseModel):
    content: str = Field(..., min_length=1)
    contact: str | None = None


class FeedbackUpdate(BaseModel):
    content: str | None = None
    contact: str | None = None
    status: str | None = None
    reply: str | None = None


class FeedbackOut(FeedbackBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int | None = None
    reply: str | None = None
    reply_at: datetime | None = None
    created_at: datetime
    updated_at: datetime
